### PR TITLE
make ingester reconnects backoff for all cases and also back off exponetially

### PR DESCRIPTION
**Pull request by kris-watts-gravwell**
PLEASE review carefully, reconnect logic changes are scary

### CODE REVIEW

File: ingest/muxer.go [hunk 1]
```diff
@@ -1438,7 +1439,28 @@ func isFatalConnError(err error) bool {
 	return false
 }
 
+func (im *IngestMuxer) quitableSleep(dur time.Duration) (quit bool) {
+	select {
+	case _ = <-time.After(defaultRetryTime):
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1444]: should this be using dur instead of the default constant?

File: ingest/muxer.go [hunk 2]
```diff
@@ -1466,9 +1488,8 @@ loop:
 				log.KV("ingesteruuid", im.uuid),
 				log.KVErr(err))
 			//non-fatal, sleep and continue
-			select {
-			case _ = <-time.After(defaultRetryTime):
-			case _ = <-im.dieChan:
+			retryDuration = backoff(retryDuration, maxRetryTime)
+			if im.quitableSleep(defaultRetryTime) {
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1492]: should this be using retryDuration instead of the default constant?

File: ingest/muxer.go [hunk 3]
```diff
@@ -1496,18 +1517,30 @@ loop:
 				log.KV("version", version.GetVersion()),
 				log.KV("ingesteruuid", im.uuid),
 				log.KVErr(err))
+			//non-fatal, sleep and continue
+			retryDuration = backoff(retryDuration, maxRetryTime)
+			if im.quitableSleep(defaultRetryTime) {
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1522]: should this be using retryDuration instead of the default constant?

File: ingest/muxer.go [hunk 4]
```diff
@@ -1496,18 +1517,30 @@ loop:
...
 			im.Error("Failed to identify ingester",
 				log.KV("indexer", tgt.Address),
 				log.KV("ingester", im.name),
 				log.KV("version", version.GetVersion()),
 				log.KV("ingesteruuid", im.uuid),
-				log.KVErr(err))
+				log.KVErr(lerr))
+			//non-fatal, sleep and continue
+			retryDuration = backoff(retryDuration, maxRetryTime)
+			if im.quitableSleep(defaultRetryTime) {
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1540]: same

File: ingest/muxer.go [hunk 5]
```diff
@@ -1517,15 +1550,21 @@ loop:
...
 					log.KV("indexer", tgt.Address),
 					log.KV("ingester", im.name),
 					log.KV("version", version.GetVersion()),
 					log.KV("ingesteruuid", im.uuid),
-					log.KVErr(err))
+					log.KVErr(lerr))
 				ig.Close()
+				//non-fatal, sleep and continue
+				retryDuration = backoff(retryDuration, maxRetryTime)
+				if im.quitableSleep(defaultRetryTime) {
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1564]: same

File: ingest/muxer.go [hunk 6]
```diff
@@ -1536,17 +1575,23 @@ loop:
...
 				log.KV("indexer", tgt.Address),
 				log.KV("ingester", im.name),
 				log.KV("version", version.GetVersion()),
 				log.KV("ingesteruuid", im.uuid),
-				log.KVErr(err))
+				log.KVErr(lerr))
 			ig.Close()
+			//non-fatal, sleep and continue
+			retryDuration = backoff(retryDuration, maxRetryTime)
+			if im.quitableSleep(defaultRetryTime) {
```
Comment (david-fritz-gravwell) [inline ingest/muxer.go:1591]: same

**Review by david-fritz-gravwell — CHANGES_REQUESTED** [review]
ok t

File: ingest/muxer.go (hunk 1 above)
Comment (kris-watts-gravwell) [reply ingest/muxer.go]: OOF.  good catch!

File: ingest/muxer.go (hunk 5 above)
Comment (kris-watts-gravwell) [reply ingest/muxer.go]: good catch

**Comment by kris-watts-gravwell** [comment]
@david-fritz-gravwell PTAL

**Review by david-fritz-gravwell — APPROVED** [review]
LGTM

**Pull request merged**